File: backend/src/main/java/co/icesi/TallerJPA/model/Progress.java
package co.icesi.TallerJPA.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.GenerationType;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "progress")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Progress {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private LocalDate date;
    private String periodType;
    private Integer repetitions;
    private Integer timeMinutes;
    private String effortLevel;
    private BigDecimal weight;
    private String notes;
    private LocalDateTime createdAt;
    
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "routine_id")
    private Routine routine;

    @ManyToOne
    @JoinColumn(name = "exercise_id")
    private Exercise exercise;

    @PrePersist
    void prePersist() {
        createdAt = LocalDateTime.now();
    }
}

File: _project_review/proyecto-final-front-errorcapa8-dev/front/bienestar-app/src/types/progress.ts
export interface ProgressEntry {
  id: number;
  date: string;
  periodType?: string | null;
  repetitions?: number | null;
  timeMinutes?: number | null;
  effortLevel: string;
  weight?: number | null;
  notes?: string | null;
  routineId?: number | null;
  routineName?: string | null;
  exerciseId?: number | null;
  exerciseName?: string | null;
}

export interface ProgressRequest {
  date: string;
  periodType?: string;
  repetitions?: number;
  timeMinutes?: number;
  effortLevel: string;
  weight?: number;
  notes?: string;
  routineId?: number;
  exerciseId?: number;
}
